### PR TITLE
Added Email Validation

### PR DESCRIPTION
**Pull request by sofiapm**
Added two new api for email validation (important to avoid bounces):

- GET /address/validate
- GET /address/parse

Mailgun documentation on this: https://documentation.mailgun.com/api-email-validation.html
Add tests for them and added information on README and Docs.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,6 +41,7 @@
   },
   "scripts": {
     "test": "mocha",
+    "debugtest": "mocha --debug-brk $npm_package_options_mocha",
```
Comment (jcurtis) [inline package.json]: 💥

Comment (sofiapm) [inline package.json]: Thanks for taking the time to review the PR. Could you elaborate on what is wrong with the task `debugtest`? Not sure whats wrong there...

Comment (atrites) [inline package.json]: Looks like it was just a temporary job that may have been left in by mistake. If it's intended then maybe add the script to the docs.

Comment (jcurtis) [inline package.json]: Yeah I think it just clutters the package.json file

You can run with debug flags manually with

`npm test -- --debug-brk`

Comment (sofiapm) [inline package.json]: Oh ok. Yeah you're right.

I'll remove it then.

Thanks

File: README.md [hunk 2]
```diff
@@ -363,6 +363,47 @@ function router(app) {
 }
 ```
 
+## Email Addresses validation
+
+Please check Mailgun [email validation documentation](https://documentation.mailgun.com/api-email-validation.html) for responses details.
+
```
Comment (bojand) [inline README.md]: Can we add a note specifying that these routes require the Mailgun public API key please.

**Comment by bojand** [comment]
Hello, thanks for the thorough PR! I agree that this would be useful functionality. Presently with this PR the clients have to create a different instance of `mailgun-js` object to just run these methods. I think a slightly better solution might be to include additional `publicApiKey` param to the constructor / initialization of the lib. And if present we can use that specifically for the auth for these requests. 

We can add a utility method in `mailgun.js` something like `getRequestOptions`:

```js
Mailgun.prototype.getRequestOptions = function (resource) {
  var o = this.options;

  //filter out API calls that do not require a domain specified
  if (resource.indexOf('/address') >= 0 && this.publicApiKey)
    var copy = Object.assign({}, this.options);
    copy.auth = [this.username, this.publicApiKey].join(':');
    p = copy;
  }
 
  return o;
};
```

and then call that function to get the request options in the `request` on [line 72](https://github.com/bojand/mailgun-js/blob/master/lib/mailgun.js#L72)

What does everyone think?

**Comment by sofiapm** [comment]
Hi, yes I agree.
I'll probably be able to fix that in the next couple of days.

**Comment by bojand** [comment]
Published as `0.10.1` (due to a npm publish mishap).
Will update website soon a bit later.

**Pull request merged**